addons/survey/controllers/main.py

```diff
@@ -732,35 +727,113 @@ def _get_user_input_domain(self, survey, line_filter_domain, **post):
+
+        # Char_box, Text_box, Numerical_box, Date, Datetime filters
+        if user_input_lines_ids:
+            user_input_lines = request.env['survey.user_input.line'].browse(user_input_lines_ids)
+            for input_line in user_input_lines:
+                value, operator = input_line._get_value_with_operator()
+                user_input_line_subdomains += input_line._get_matching_answer_domain(operator, value)
+                search_filters.append(self._prepare_search_filter('L', input_line, value))
+
+        # Compute base domain
+        user_input_domain = self._get_results_page_user_input_domain(survey, **post)
+
+        # Add filters domain to the base domain
+        if user_input_line_subdomains:
+            all_required_lines_domains = [
+                [('user_input_line_ids', 'in', request.env['survey.user_input.line'].sudo()._search(subdomain))]
+                for subdomain in user_input_line_subdomains
+            ]
+            user_input_domain = expression.AND([user_input_domain, *all_required_lines_domains])
+
+        # Get the matching user input lines
+        user_inputs_query = request.env['survey.user_input'].sudo()._search(user_input_domain)
+        user_input_lines = request.env['survey.user_input.line'].search([('user_input_id', 'in', user_inputs_query)])
 
         return user_input_lines, search_filters
+
+    def _get_filters_from_post(self, post):
+        """ Extract the filters from post depending on the model that needs to be called to retrieve the filtered answer data.
+        Simple choice and multiple choice question types are mapped onto empty row_id.
+        Input/output example with respectively matrix, simple_choice and char_box filters:
+            input: 'A,1,24|A,0,13|L,0,36'
+            output:
+                answer_by_column: {24: [1], 13: []}
+                user_input_lines_ids: [36]
+
+        * Model short key = 'A' : Match a `survey.question.answer` record (simple_choice, multiple_choice, matrix)
+        * Model short key = 'L' : Match a `survey.user_input.line` record (char_box, text_box, numerical_box, date, datetime)
+        :rtype: (collections.defaultdict[int, list[int]], list[int])
+        """
+        answer_by_column = defaultdict(list)
+        user_input_lines_ids = []
+
+        for data in post.get('filters', '').split('|'):
+            if not data:
+                break
+            model_short_key, row_id, answer_id = data.split(',')
+            row_id, answer_id = int(row_id), int(answer_id)
+            if model_short_key == 'A':
+                if row_id:
+                    answer_by_column[answer_id].append(row_id)
+                else:
+                    answer_by_column[answer_id] = []
+            elif model_short_key == 'L' and not row_id:
+                user_input_lines_ids.append(answer_id)
+
+        return answer_by_column, user_input_lines_ids
+
+    def _prepare_search_filter(self, model_short_key, record, value, row_id=0):
```

Comment:
Actually I would split "_prepare_search_filter" into "_prepare_search_filter_answer" and "_prepare_search_filter_input_line".

And move calls to "_get_filter_value" over there for both answers and user_input_lines.

For the operator part, I would create a method "_get_answer_value_domain_operator" the same way you did with "_get_answer_value_field" with a bunch of if / else, and call it from within "_get_matching_answer_domain" as it's only used there it seems.
(If only used once at the end, you can remove the method and copy/paste the code, same rule as previously).

For "get_filter_value", I would remove the method and put the code in "_prepare_search_filter_answer" and simplify, the current construct with 2 f-strings is really hard to read, I'm honestly not sure what it does.

Current feeling is that code tries to do some clever constructs to avoid repetition but it's hard to follow, would prefer clear calls even if it's a bit more code in the end :slightly_smiling_face: